File: backend/app/schemas/audit_logs.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class AuditLogRead(BaseModel):
    id: UUID
    actor_id: UUID | None
    action: str
    target_type: str
    target_id: str
    before_state: dict[str, object]
    after_state: dict[str, object]
    ip_address: str | None
    user_agent: str | None
    extra: dict[str, object]
    request_id: str | None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class AuditLogListRead(BaseModel):
    items: list[AuditLogRead]
    page: int
    page_size: int
    total: int

File: frontend/src/types/auditLogs.ts
export interface AuditLogRead {
  action: string;
  actor_id: string | null;
  after_state: Record<string, unknown>;
  before_state: Record<string, unknown>;
  created_at: string;
  extra: Record<string, unknown>;
  id: string;
  ip_address: string | null;
  request_id: string | null;
  target_id: string;
  target_type: string;
  updated_at: string;
  user_agent: string | null;
}

export interface AuditLogListRead {
  items: AuditLogRead[];
  page: number;
  page_size: number;
  total: number;
}

export interface AuditLogListQuery {
  action?: string;
  actorId?: string;
  createdFrom?: string;
  createdTo?: string;
  page?: number;
  pageSize?: number;
  targetId?: string;
  targetType?: string;
}
